lib/Backends/Habana/Habana.cpp

```diff
@@ -422,6 +435,26 @@ static bool allowsPartialInput(const Node *src, const Node *dst) {
   return false;
 }
 
+/// \returns true if \p dst allows \p src to be converted from 64 to 32 bits.
+static bool allows64To32Downcast(const Node *src, const Node *dst) {
```

Comment:
This would improve perf of SLS for all backends. Can this be a pass that we do higher in the stack instead of buried in/limited to one backend?

Reply:
Right -- I think really this is really something that can be easily fixed at the model layer.  Doing it in the Habana backend is a short-term hack to help figure out the perf implications of 64-vs-32 bit indices.

I'm uncertain what the right way to provide this for all backends is... maybe insert some 64-to-32 casts and have that handled by the CPU backend?  (Of course, this particular transformation is a pessimization for CPU, but should be a win for ASICs and maybe GPUs)